### ISSUE TITLE
Keyboard pops up on channel change on mobile

### ISSUE THREAD

**Issue by YaManicKill** [thelounge/lounge#758]
Because we now put focus to the text input, it means that my keyboard is popping up everytime I change channel on Android Chrome.

This seems less than ideal.

**Comment by astorije** [comment]
That's weird, [focusing is supposed to be enabled on mobile only](https://github.com/thelounge/lounge/blob/a6f0dfbe4761021060e14edd189baab030a4ae06/client/js/lounge.js#L933-L938) (in a very much not optimal way, that I agree, but our options are limited 😕).

What is the resolution of your device?

**Comment by YaManicKill** [comment]
Ah, maybe it is due to the fact that I have my device as a higher density, meaning that it appears to be a higher resolution to sites. Could we change it so that it works it based on if we are a touch screen rather than the size of device?

**Comment by YaManicKill** [comment]
Confirmed, it's because I use the Android setting to make my display more compact.

So I guess we have a couple of options with this:

1. Up the resolution that we stop doing this. This would then work for high-resolution phones and tablets, but might catch some low-resolution displays
2. Do it for touch screen devices. This would do all phones and tablets, but would catch some touch screen laptops
3. Combination of the 2, maybe?

Any better ideas? It's obviously quite frustrating to have the keyboard pop up everytime I move room.

**Comment by xPaw** [comment]
I am seeing the same problem on OnePlus 3 with stock dpi. Screen size test is certainly not good enough.

**Comment by astorije** [comment]
I think we reached a point where screen size indeed is not good enough.

What if, instead of relying on that (which is simply an extension of media queries, we were doing like GitHub for example, detect the device type and save this as part of the settings?
So when user loads the app for the first time, we can detect what kind of device it is, desktop / mobile (maybe tablet as well if necessary?). That way, we can have a more unified way to make decisions in the code / CSS. And add an extra select box in the client settings that persists in localStorage in case we got it wrong, or if it's a device that acts as both depending on the user, etc.

Does this make any kind of sense?
